// stimulsoft-viewer-angular/esm2020/lib/menu/digital-signature-menu.component copy.mjs
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/model.service";
import * as i2 from "../services/export.service";
import * as i3 from "@angular/common";
export class DigitalSignatureMenuComponent {
    constructor(model, exportService) {
        this.model = model;
        this.exportService = exportService;
    }
    ngOnInit() { }
}
DigitalSignatureMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: DigitalSignatureMenuComponent, deps: [{ token: i1.ModelService }, { token: i2.ExportService }], target: i0.ɵɵFactoryTarget.Component });
DigitalSignatureMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: DigitalSignatureMenuComponent, selector: "sti-digital-signature-menu", ngImport: i0, template: `
    <div *ngFor="let item of model.pdfSecurityCertificates" class="stiJsViewerMenuStandartItem stiJsViewerMenuStandartItemSelected" style="height: auto; line-height: 1.3;">
      <table class="stiJsViewerClearAllStyles" cellpadding="0" cellspacing="0" style="height: 100%; width: 100%;">
        <tbody>
          <tr class="stiJsViewerClearAllStyles">
            <td class="stiJsViewerClearAllStyles" style="padding: 8px 20px 8px 8px; text-align: left; white-space: nowrap;">
            </td>
          </tr>
        </tbody>
      </table>
    </div>
`, isInline: true, directives: [{ type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: DigitalSignatureMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'sti-digital-signature-menu',
                    template: `
    <div *ngFor="let item of model.pdfSecurityCertificates" class="stiJsViewerMenuStandartItem stiJsViewerMenuStandartItemSelected" style="height: auto; line-height: 1.3;">
      <table class="stiJsViewerClearAllStyles" cellpadding="0" cellspacing="0" style="height: 100%; width: 100%;">
        <tbody>
          <tr class="stiJsViewerClearAllStyles">
            <td class="stiJsViewerClearAllStyles" style="padding: 8px 20px 8px 8px; text-align: left; white-space: nowrap;">
            </td>
          </tr>
        </tbody>
      </table>
    </div>
`
                }]
        }], ctorParameters: function () { return [{ type: i1.ModelService }, { type: i2.ExportService }]; } });
