// stimulsoft-viewer-angular/esm2020/lib/menu/vertical-menu.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./vertical-menu-item.component";
import * as i2 from "@angular/common";
export class VerticalMenuComponent {
    constructor() { }
    ngOnInit() { }
    action(item) {
        this.menu?.action?.emit(item);
    }
}
VerticalMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: VerticalMenuComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
VerticalMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: VerticalMenuComponent, selector: "sti-vertical-menu", inputs: { menu: "menu" }, ngImport: i0, template: `
    <sti-vertical-menu-item *ngFor="let item of menu.verticalItems" [item]="item" [styleName]="menu.itemStyleName" (action)="action(item)">
    </sti-vertical-menu-item>
  `, isInline: true, components: [{ type: i1.VerticalMenuItemComponent, selector: "sti-vertical-menu-item", inputs: ["styleName", "item", "imageSizesWidth", "imageSizesHeight", "enabled"], outputs: ["action"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: VerticalMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'sti-vertical-menu',
                    template: `
    <sti-vertical-menu-item *ngFor="let item of menu.verticalItems" [item]="item" [styleName]="menu.itemStyleName" (action)="action(item)">
    </sti-vertical-menu-item>
  `
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { menu: [{
                type: Input
            }] } });
